File: sdk/v1.0.67/python/mediaviz_sdk/ai_model_credits.py
from __future__ import annotations
from typing import Any
from urllib.parse import quote, urlencode
import httpx

from .errors import handle_response


class AiModelCredits:
    def __init__(self, ctx) -> None:
        self._ctx = ctx

    def get_model_credit_relationship(self, model_name: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/model_credit/' + quote(str(model_name), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

File: sdk/v1.9.0-dev.117/react_native/ai_model_credits.js
export class AiModelCredits {
  constructor(ctx) { this._ctx = ctx; }

  async getModelCreditRelationship(modelName) {
    this._ctx.requireTokens();
    const path = `/api/v1/model_credit/${encodeURIComponent(modelName)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }
}
